ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthCreateAccessList.java

```diff
@@ -0,0 +1,191 @@
+import java.util.function.Function;
+
+public class EthCreateAccessList extends EthEstimateGas {
+
+  public EthCreateAccessList(
+      final BlockchainQueries blockchainQueries, final TransactionSimulator transactionSimulator) {
+    super(blockchainQueries, transactionSimulator);
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ETH_CREATE_ACCESS_LIST.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    final JsonCallParameter jsonCallParameter = validateAndGetCallParams(requestContext);
+    final BlockHeader blockHeader = blockHeader();
+    final Optional<JsonRpcError> jsonRpcError = validateBlockHeader(blockHeader);
+    if (jsonRpcError.isPresent()) {
+      return errorResponse(requestContext, jsonRpcError.get());
+    }
+    final AccessListSimulatorResult maybeResult =
+        maybeProcessTransaction(jsonCallParameter, blockHeader);
+    // if the call accessList is same than the simulation result, calculate gas and return
+    if (isAccessListResultValid(jsonCallParameter, maybeResult.getTracer())) {
+      return createResponse(requestContext, maybeResult);
+    } else { // process transaction with the simulation accessList result
+      final AccessListSimulatorResult result =
+          processTransactionWithAccessList(
+              jsonCallParameter, blockHeader, maybeResult.getTracer().getAccessList());
+      return createResponse(requestContext, result);
+    }
+  }
+
+  private Optional<JsonRpcError> validateBlockHeader(final BlockHeader blockHeader) {
+    if (blockHeader == null) {
+      return Optional.of(JsonRpcError.INTERNAL_ERROR);
+    }
+    if (!blockchainQueries
+        .getWorldStateArchive()
+        .isWorldStateAvailable(blockHeader.getStateRoot(), blockHeader.getHash())) {
+      return Optional.of(JsonRpcError.WORLD_STATE_UNAVAILABLE);
+    }
+    return Optional.empty();
+  }
+
+  private JsonRpcResponse createResponse(
+      final JsonRpcRequestContext requestContext, final AccessListSimulatorResult result) {
+    return result
+        .getResult()
+        .map(createResponse(requestContext, result.getTracer()))
+        .orElse(errorResponse(requestContext, JsonRpcError.INTERNAL_ERROR));
+  }
+
+  private TransactionValidationParams transactionValidationParams(
+      final boolean isAllowExceedingBalance) {
+    return ImmutableTransactionValidationParams.builder()
+        .from(TransactionValidationParams.transactionSimulator())
+        .isAllowExceedingBalance(isAllowExceedingBalance)
+        .build();
+  }
+
+  private boolean isAccessListResultValid(
+      final JsonCallParameter parameters, final AccessListOperationTracer tracer) {
+    if (tracer.getAccessList().isEmpty()) {
+      return true; // did not access any storage
+    } else {
+      if (parameters.getAccessList().isEmpty()) {
+        return false;
+      }
+      return parameters.getAccessList().get().isEmpty()
```

Comment:
at this point this is guaranteed false, no? already returned if it was true on line 107?
 parameters.getAccessList().get().isEmpty()

Reply:
Lines 107: isEmpty is testing  the Optional<List>. 
Line 109: isEmpty is testing  the list itself.

Reply:
I have inverted the logic and removed the list test since it is actually redundant. Also added some comments for clarity.